packages/js/src/redirects/components/table-rows.js

```diff
@@ -0,0 +1,155 @@
+import { Table, Checkbox, DropdownMenu, SelectField, TextField, Button } from "@yoast/ui-library";
+import {  useFormikContext } from "formik";
+import FormikWithErrorField from "../../shared-admin/components/form/formik-with-error-field";
+import { FormikValueChangeField } from "../../shared-admin/components/form";
+import { __ } from "@wordpress/i18n";
+import { REDIRECT_TYPE_SHORTS_OPTIONS } from "../constants";
+import { useCallback } from "@wordpress/element";
+
+/**
+ * TableRows — renders each row of the redirects table, including editable rows.
+ *
+ * This component displays a list of redirect entries with the ability to:
+ * - Select individual redirects via checkboxes
+ * - Edit a redirect inline using Formik fields
+ * - Trigger a dropdown menu for "Edit" and "Delete" actions
+ *
+ * @param {Array<Object>} sortedRedirects - List of redirect entries to display.
+ * @param {Array<string>} selectedRedirects - Array of selected redirect IDs.
+ * @param {Function} onToggleSelect - Handler for checkbox selection toggle.
+ * @param {Function} handleDeleteModal - Function to toggle the delete confirmation modal.
+ * @param {Object} selectedRedirect - Redirect currently being edited.
+ * @param {Function} handleEditClick - Handler to activate edit mode or cancel edit.
+ * @param {Function} setSelectedDeleteRedirect - Sets the redirect to delete (used in modal).
+ *
+ * @returns {JSX.Element} Table rows for each redirect, including editable and action-enabled rows.
+*/
+export const TableRows =   (
+	{
+		sortedRedirects,
+		selectedRedirects,
+		onToggleSelect,
+		handleDeleteModal,
+		selectedRedirect,
+		handleEditClick,
+		setSelectedDeleteRedirect,
+	}
+) => {
+	const { isSubmitting } = useFormikContext();
+
+	const editClickHandler = useCallback(
+		( type, target, origin ) => () => {
+			handleEditClick( { type, target, origin } );
+		},
+		[ handleEditClick ]
+	);
+
+	const onDelete = useCallback( ( values ) => () => {
+		setSelectedDeleteRedirect( values );
+		handleDeleteModal();
+	}, [ handleDeleteModal, setSelectedDeleteRedirect ] );
+
+	return sortedRedirects?.map( ( { id, type, target, origin } ) => {
+		const isEditing = selectedRedirect.origin === origin;
+
+		if ( isEditing ) {
+			return (
+				<Table.Row key={ id } className="yst-cell-edit-redirect">
+					<Table.Cell>
+						<div className="yst-flex yst-items-center">
+							<Checkbox
```

Comment:
Actually label is required in for this component. I think that is something we should fix in the `ui-library` level. That will make unnecessary the css for the label. I can create a PR for that.